project/ThreeMonth03/README.md

```diff
@@ -0,0 +1,121 @@
+# NSD Proposal
+
+## Basic Information
+
+I will implement conjugate gradient method as my final project.
+
+Project URL: [https://github.com/ThreeMonth03/Conjugate-Gradient-Method-library](https://github.com/ThreeMonth03/Conjugate-Gradient-Method-library)
+
+
+## Problem to Solve
+
+I want to solve the global minima of the quadratic function by searching the numerical solution. One of the algorithm that can solve the problem is conjugate gradient method.
+
+## Conjugate gradient method
+### Introduction
+
+Conjugate gradient method is a gradient-descent-based method that can find the minima of the quadratic function in the hyper-dimensional space. 
+
+Given the quadratic function f(x):
+
+$$\begin{aligned}
+f(x) = \frac{1}{2}x^{T}Hx + b^{T}x + c
+\end{aligned}
+$$
+
+The algorithm of conjugate gradient method can find 
+$$ x_{min} = \underset{x}{\mathrm{argmin}} \quad f(x)$$
+
+Compared with the steepest gradient method, the minima could be found in the finite step by conjugate gradient method in theory (if we don't consider the floating-point error and ill-conditioned), and the convergence iteration of conjugate gradient method is less than those of steepest gradient method.
+
+### Algorithm
+Given the function f(x):
+
+$$\begin{aligned}
+f(x) = \frac{1}{2}x^{T}Hx + b^{T}x + c
+\end{aligned}
+$$
+
+And the algorithm is:
+
+
+
+$$\begin{aligned}  
+& Input : Point_{initial}=x_{0} , Iteration = n,Coefficient = H, b\\\\    
+& Initialize : s_{-1}= 0, \beta_{-1} = 1, g_{0} = g(x_{0}) = Hx_{0} + b, i = 0\\\\       
+& While((i \< n) \ and  \ (x_{i} \neq x_{i-1})):\\\\    
+& \quad \quad \quad \quad s_{i} = -g_{i} + \beta_{i-1}s_{i-1}\\\\  
+& \quad \quad \quad \quad a_{i} = \frac{g_{i}^{T}s_{i}}{s_{i}^{T}Hs_{i}}\\\\  
+& \quad \quad \quad \quad x_{i+1} = x_{i} + a_{i}s_{i}\\\\  
+& \quad \quad \quad \quad g_{i+1} = g(x_{i}) = Hx_{i} + b\\\\  
+& \quad \quad \quad \quad b_{i} = \frac{\lVert g_{i+1} \rVert^{2}}{\lVert g_{i} \rVert^{2}}\\\\  
+& \quad \quad \quad \quad i = i+1\\\\  
+\end{aligned}  
+$$
+
+## Prospective Users
+
+Conjugate gradient method is pretty suitable for solving the regression problem. The algorithm can help the researchers working in deep learning, optimization and social science. 
+
+## System Architecture
+
+The above algorithm will be encapsulated by c++, and the algorithm also could be called by python with pybind11.
+
+### Constraints
+
+The system assumes that the input $H$ is positive-definite matrix and well-conditioned. 
+The quadratic function exists a global minima if and only if the matrix $H$ is positive-definite. Additionally, the convergence speed is pretty fast only when $H$ is well-conditioned. 
```

Comment:
There will be little for discussions in your presentation if everything is positive-definite.  (Solution of positive-definite extrema usually does not need a numerical method.)

To make a decent presentation, you probably will need to handle some ill-behaviors.  There should be plenty of literature discussing it.  Pick simple ones to proceed.

Reply:
By the way, I ignore a strict condition of $H$, that $H$ should also be a symmetric matrix.
To be frank, it seems that the application of conjugate gradient method is few. 
Do you suggest that I should find another topic?

Reply:
I did not suggest you to change a numerical method.  The numerical method is not the point.  The point is how to design and implement the software system surrounding the numerical method.  All numerical methods have limitation.  The software system is to make them useful to the most extent.